### PR TITLE
feat(ci): add pre-push hook running unit tests for changed packages

### PR DESCRIPTION
**Pull request by Kiran01bm**
## Summary

Adds a `.githooks/pre-push` hook that runs `go test -race` for the packages
affected by the commits being pushed. It gives fast local feedback that mirrors
the CI unit job, scoped to changed packages so it stays quick. CI remains the
authoritative gate and runs the full suite — this hook does not replace it.

## How it works

The hook diffs the pushed commit range for `*.go` changes, maps them to package
directories that still exist, and tests only those:
```
pushed range --> changed *.go --> existing package dirs --> go test -race (only those)
|
none --> skip
```
Edge cases handled:

- New-branch push: range is the merge-base against the remote's default branch.
- Branch deletion / no-op push: skipped.
- Removed package directory: dropped from the test set.

GOROOT/PATH resolution mirrors the existing pre-commit hook, and `make setup`
already enables the hook via `core.hooksPath`.

### CODE REVIEW

File: .githooks/pre-push [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+            range_base="${local_sha}~1"
+        fi
+    else
+        range_base="$remote_sha"
+    fi
+
+    files=$(git diff --name-only --diff-filter=d "$range_base" "$local_sha" -- '*.go' 2>/dev/null || true)
+    if [ -n "$files" ]; then
+        changed_go_files="$changed_go_files
+$files"
```
[the remark this answers is not in the document]
Comment (Kiran01bm) [reply .githooks/pre-push]: fixed in https://github.com/block/schemabot/pull/394/changes/02d50cfbe35299890bda58c50d89eb9d21c93f52

**Review by aparajon — APPROVED** [review]
